drivers/gpu/drm/rp1/rp1-dsi/rp1_dsi.c

```diff
@@ -54,11 +54,18 @@ static void rp1_dsi_bridge_pre_enable(struct drm_bridge *bridge,
 	struct rp1_dsi *dsi = bridge_to_rp1_dsi(bridge);
 
 	rp1dsi_dsi_setup(dsi, &dsi->pipe.crtc.state->adjusted_mode);
+	dsi->dsi_running = true;
 }
 
 static void rp1_dsi_bridge_enable(struct drm_bridge *bridge,
 				  struct drm_bridge_state *old_state)
 {
+	struct rp1_dsi *dsi = bridge_to_rp1_dsi(bridge);
+
+	if (!dsi->dsi_running) {
+		rp1dsi_dsi_setup(dsi, &dsi->pipe.crtc.state->adjusted_mode);
+		dsi->dsi_running = true;
+	}
```

Comment:
How are you thinking you can get to `enable` without `pre_enable` being called?

Reply:
Ah, I wasn't sure if it was mandatory. BTW it would be nice if pre-enable was called before any DSI commands got sent.

Reply:
Gone.